src/arviz_base/converters.py

```diff
@@ -1,14 +1,23 @@
 """Generalistic converters."""
 
 import numpy as np
+import pandas as pd
 import xarray as xr
 from datatree import DataTree, open_datatree
 
 from arviz_base.base import dict_to_dataset
+from arviz_base.labels import BaseLabeller
 from arviz_base.rcparams import rcParams
+from arviz_base.sel_utils import xarray_sel_iter
 from arviz_base.utils import _var_names
 
-__all__ = ["convert_to_datatree", "convert_to_dataset", "extract"]
+__all__ = [
+    "convert_to_datatree",
+    "convert_to_dataset",
+    "extract",
+    "to_labelled_stacked_da",
+    "to_labelled_stacked_df",
+]
```

Comment:
My opinion for naming is typically to use modules (or classes) and methods therein to write a sentence (or command) that someone can read as a sentence. For instance, the difference between

```python
azb.converter.convert_to_datatree(...)
```

and

```python
azb.convert.to_datatree(...)
```

is (written as sentences in my head as)

- ArviZ base converter convert to datatree (...).
- ArviZ base convert to datatree (...).

Semantically they both mean the same thing, but the second sentence makes more sense grammatically. Things like `da`, `ds`, `df` and now `dt` are shortened versions of objects people are familiar with, so the `to_labelled_stacked_df` does make sense to me, and I like it.

If we are discussing an architecture for converting a dataset, I (as a user/reader of code) would think it is easier to read (for me at least) `azb.convert.to_df(ds=ds, sample_dims=None, stacked=True, labeller=...)`
than `azb.converters.to_labelled_stacked_df(ds=ds, sample_dims=None, stacked=False, labeller=...)` since the former is closer to what I would read as this sentence;

> ArviZ base convert to dataframe (using this dataset, with these sample dimensions, and ensure it is stacked, using this labeller).

This is **purely** an opinion though, and I think the way you have written it makes perfect sense.

I only thing I would suggest is to make things consistent, so maybe choosing how you want the methods to read. For example using all abbreviations or not (_eg_ `dataset` _vs_ `ds` or `df` _vs_ `dataframe`).

| original               | abbreviated | long version |
| ---------------------- | ----------- | ------------ |
| convert_to_datatree    | to_dt       | to_datatree  |
| convert_to_dataset     | to_ds       | to_dataset   |
| to_labelled_stacked_da | to_da       | to_dataarray |
| to_labelled_stacked_df | to_df       | to_dataframe |